### PR TITLE
util/metrics: add threads IO metrics

### PR DESCRIPTION
**Pull request by overvenus**
Collects threads IO usage.

screenshot from 2018-07-20 12-39-43

### CODE REVIEW

**Comment by breeswish** [comment]
OK.

**Comment by overvenus** [comment]
PTAL, thanks!

File: src/util/metrics/threads_linux.rs [hunk 1]
```diff
@@ -228,6 +272,67 @@ fn get_thread_stat_internal(tid: pid_t, stat: &str) -> Result<Stat> {
...
+        let mut io = String::new();
+        fs::File::open(format!("/proc/{}/task/{}/io", pid, tid))
+            .and_then(|mut f| f.read_to_string(&mut io))?;
+        get_thread_io_internal(&io)
+    }
+}
+
+// Extracted from `Io::collect`, for test purpose.
+fn get_thread_io_internal(io: &str) -> Result<Io> {
+    let read_bytes = io
```
Comment (breeswish) [inline src/util/metrics/threads_linux.rs]: please extract a `let lines = io.lines()` to avoid repeat line split. rest LGTM.

Comment (overvenus) [reply src/util/metrics/threads_linux.rs]: I tried, and found out it became less straightforward, the link below is an example.
It's not performance critical, so I think this is fine.

https://play.rust-lang.org/?gist=88f63c8d5a8000bb683a26bc2ec68d72&version=stable&mode=debug&edition=2015

```rust
#![allow(unused)]
fn main() {
let a = [1, 2, 3];

let mut iter = a.iter();

assert_eq!(iter.nth(0), Some(&1));
assert_eq!(iter.nth(1), Some(&2));
}

thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `Some(3)`,
 right: `Some(2)`', src/main.rs:8:1
```

**Pull request merged**